fuel-vm/src/interpreter/executors/main.rs

```diff
@@ -110,6 +113,32 @@ impl<T> Interpreter<PredicateStorage, T> {
         Self::run_predicate(PredicateRunKind::Verifying(tx), balances, params, gas_costs)
     }
 
+    /// Initialize the VM with the provided transaction and check all predicates defined
+    /// in the inputs in parallel.
+    ///
+    /// The storage provider is not used since contract opcodes are not allowed for
+    /// predicates.
+    pub async fn check_predicates_async<Tx, E>(
+        checked: Checked<Tx>,
+        params: ConsensusParameters,
+        gas_costs: GasCosts,
+    ) -> Result<(PredicatesChecked, Checked<Tx>), PredicateVerificationFailed>
+    where
+        Tx: ExecutableTransaction + Send + 'static,
+        <Tx as IntoChecked>::Metadata: CheckedMetadata,
+        E: ParallelExecutor
+            + ParallelExecutor<TaskResult = Result<Word, PredicateVerificationFailed>>,
+    {
+        let tx = checked.transaction().clone();
```

Comment:
You don't need to clone it at this point. You will clone it later

Reply:
done